parquet-thrift/src/main/java/parquet/thrift/ThriftRecordConverter.java

```diff
@@ -843,7 +843,9 @@ public T getCurrentRecord() {
       rootEvents.clear();
       return thriftReader.readOneRecord(protocol);
     } catch (TException e) {
-      throw new ParquetDecodingException("Could not read thrift object from protocol", e);
+      protocol.clear();
+      rootEvents.clear();
```

Comment:
I wanted to ask you that -- also is `skipCurrentRecord` doing the right thing or should it clear both as well?

I confirmed that you need to clear protocol for this to work. I think it was formerly never a problem because this exception was fatal so it didn't matter.